src/path_manager.c

```diff
@@ -105,86 +105,158 @@ static char const *get_pm_name(void const *data, size_t len)
 {
         struct l_genl_attr attr;
         if (!l_genl_attr_init(&attr, msg)) {
                 l_error("Unable to initialize genl attribute");
                 return;
         }
 
-        /*
-          Payload:
-              Token
-              Local address
-              Local port
-              Remote address
-              Remote port
-              Path management strategy (optional)
-        */
-
-        mptcpd_token_t  const *token       = NULL;
-        struct in_addr  const *laddr4      = NULL;
-        struct in_addr  const *raddr4      = NULL;
-        struct in6_addr const *laddr6      = NULL;
-        struct in6_addr const *raddr6      = NULL;
-        in_port_t       const *local_port  = NULL;
-        in_port_t       const *remote_port = NULL;
-        char            const *pm_name     = NULL;
-
         uint16_t type;
         uint16_t len;
         void const *data = NULL;
 
         while (l_genl_attr_next(&attr, &type, &len, &data)) {
                 switch (type) {
                 case MPTCP_ATTR_TOKEN:
-                        MPTCP_GET_NL_ATTR(data, len, token);
+                        MPTCP_GET_NL_ATTR(data, len, attrs->token);
                         break;
                 case MPTCP_ATTR_SADDR4:
-                        MPTCP_GET_NL_ATTR(data, len, laddr4);
+                        MPTCP_GET_NL_ATTR(data, len, attrs->laddr4);
                         break;
                 case MPTCP_ATTR_SADDR6:
-                        MPTCP_GET_NL_ATTR(data, len, laddr6);
+                        MPTCP_GET_NL_ATTR(data, len, attrs->laddr6);
                         break;
                 case MPTCP_ATTR_SPORT:
-                        MPTCP_GET_NL_ATTR(data, len, local_port);
+                        MPTCP_GET_NL_ATTR(data, len, attrs->local_port);
                         break;
                 case MPTCP_ATTR_DADDR4:
-                        MPTCP_GET_NL_ATTR(data, len, raddr4);
+                        MPTCP_GET_NL_ATTR(data, len, attrs->raddr4);
                         break;
                 case MPTCP_ATTR_DADDR6:
-                        MPTCP_GET_NL_ATTR(data, len, raddr6);
+                        MPTCP_GET_NL_ATTR(data, len, attrs->raddr6);
                         break;
                 case MPTCP_ATTR_DPORT:
-                        MPTCP_GET_NL_ATTR(data, len, remote_port);
+                        MPTCP_GET_NL_ATTR(data, len, attrs->remote_port);
                         break;
                 case MPTCP_ATTR_FAMILY:
                 case MPTCP_ATTR_LOC_ID:
                 case MPTCP_ATTR_REM_ID:
                 case MPTCP_ATTR_BACKUP:
+                case MPTCP_ATTR_ERROR:
+                case MPTCP_ATTR_FLAGS:
+                case MPTCP_ATTR_TIMEOUT:
                 case MPTCP_ATTR_IF_IDX:
+                case MPTCP_ATTR_RESET_REASON:
+                case MPTCP_ATTR_RESET_FLAGS:
                         // Unused and ignored, at least for now.
```

Comment:
We should probably retrieve at least some of these attributes, such as `MPTCP_ATTR_IF_IDX`, `MPTCP_ATTR_{LOC,REM}_ID` and `MPTCP_ATTR_BACKUP`.